src/metabase/public_settings/premium_features.clj

```diff
@@ -119,14 +119,19 @@
   ;; the first request to finish
   (let [lock (Object.)
         f    (memoize/ttl
-              fetch-token-status*
+              (fn [token]
+                ;; this is a sanity check to make sure we can actually get the active user count BEFORE we try to
+                ;; call [[fetch-token-status*]], because `fetch-token-status*` catches Exceptions and caches even failed
+                ;; results. We were running into issues in the e2e tests where `active-user-count` was timing out realed
+                ;; to weird timeouts after restoring the app DB from a snapshot, which would cause other tests to fail
```

Comment:
```suggestion
                ;; call [[fetch-token-status*]], because `fetch-token-status*` catches Exceptions and therefore caches failed
                ;; results. We were running into issues in the e2e tests where `active-user-count` was timing out because
                ;; of weird timeouts after restoring the app DB from a snapshot, which would cause other tests to fail
```